### PR TITLE
Explain that FILETYPE_ASN1 is DER.

### PR DESCRIPTION
**Pull request by Lukasa**
A minor documentation update which satisfies my own confusion.

### CODE REVIEW

File: doc/api/crypto.rst [hunk 1]
```diff
@@ -35,7 +35,12 @@ Serialization and deserialization
 ---------------------------------
 
 The following serialization functions take one of these constants to
-determine the format:
+determine the format.
+
+:data:`FILETYPE_PEM` serializes data to a Base64-encoded encoded representation
+of the underlying ASN.1 data structure. :data:`FILETYPE_ASN1` serializes data
+to the underlying ASN.1 data structure. The format used by
+:data:`FILETYPE_ASN1` is also sometimes referred to as DER.
```
Comment (alex) [inline doc/api/crypto.rst]: I would put this under the `py:data` decleration so linkification does something sane. Perhaps even split those out into two separate declerations, each with their own description.

File: doc/api/crypto.rst [hunk 2]
```diff
@@ -35,10 +35,17 @@ Serialization and deserialization
...
 
 The following serialization functions take one of these constants to
-determine the format:
+determine the format.
 
 .. py:data:: FILETYPE_PEM
-             FILETYPE_ASN1
+
+:data:`FILETYPE_PEM` serializes data to a Base64-encoded encoded representation
+of the underlying ASN.1 data structure.
```
Comment (alex) [inline doc/api/crypto.rst]: Might want to mention the delibeters, it's not _just_ `base64` encoded.

**Comment by hynek** [comment]
Semantic newlines please.  Also please make @alex happy (how hard can that be).

**Comment by hynek** [comment]
Cooooooryyyyyyyyyy

**Comment by Lukasa** [comment]
Arg sorry I totally lost track of this!

**Comment by Lukasa** [comment]
Done for both @hynek and @alex.

**Pull request merged**